### ISSUE TITLE
Tuning Services sometimes issue TMSCON0029E: The tuning connection profile was not found

### ISSUE THREAD

**Issue by Marty850** [IBM/db2forzosdeveloperextension-about#86]
## Development environment where the bug occurred

- Db2 Developer Extension version: 1.3.4
- Editor platform
  - [x] Visual Studio Code
  - [ ] Eclipse Theia
- Editor platform version: 1.65.2
- Operating system on which your editor runs (for example, Windows 10 2004 or MacOS Catalina 10.15.7): Windows 10.0.19042.1645
- Java Version (Run `java -version` and paste the details here):   
java version "1.8.0_321"  
Java(TM) SE Runtime Environment (build 1.8.0_321-b07)  
Java HotSpot(TM) Client VM (build 25.321-b07, mixed mode)  
- Db2 for z/OS version (including function level for Db2 12): V12R1M510
- [Log files](https://ibm.github.io/db2forzosdeveloperextension-about/docs/troubleshooting/opening-an-issue-and-providing-logs) attached?: No

## Problem description

Sometimes the tuning server “forgets” about his tuning profile and the message “TMSCON0029E: The tuning connection profile was not found” is issued in VSCode. Then I restart the Tuning Server STC and after that it works. There is no message in tuning server log. I did not notice a pattern that leads into this issue. Somtimes the issue exits on the next day, somtimes it works for more then one day.
Could you please help analyzing this issue? Do you expact more information if we increase the log_level?

**Comment by mengxf123** [comment]
Hi Martin, would you please let me know when did you first discover this problem? Is it when you are trying to tune a query? What did you do before you saw this error? Like adding a new connection profile through the VSC? Is this error easy to recreate?

**Comment by Marty850** [comment]
I think the problem exists since we installed the tuning services but I didn't realize it because the task was started and stopped manually for testing and there were just a few users. Now the task runs 24/7 with system automation and more people are using it so we noticed now that there is a problem.

The message occurs some time after I click "Tune" on a selected SQL. So it occurs while the tuning process is running. We did not change anything on the connection profile before or something like that. Its difficult to recreate. Sometimes it is there, sometimes not...

**Comment by mengxf123** [comment]
Can you check the tuning options of the connections when you hit the error? Is it possible you associate the connection with an old profile that has been deleted without your awareness?  
<img width="999" alt="图片" src="https://user-images.githubusercontent.com/52191735/167587035-9a7939d1-c237-4666-8917-c68eb97ca5de.png">

**Comment by Marty850** [comment]
Now the error happened again. Tuning options look like in your screenshot. Tuning connection profile is not filled and I'm not able to select one.

GET on /tuningservice/v1/connections results in:
```
{
    "code": 200,
    "status": "success",
    "payload": {
        "result": {
            "list": []
        }
    }
}
```

A select on the table IBMTMS.PROFILE will show the two profiles.

**Comment by Marty850** [comment]
Hi @mengxf123, 
I think I found the problem. I noticed that ther was an IPL yesterday evening.

z/OS LPAR was IPLed AT 20.18.28 ON 05/12/2022
DB2 DDF was available at 20:23:39
Tuning services were started at 20.21.40

So DB2 DDF was not available when tuning services were started. Maybe we can solve that by system automation but I think tuning services should also be able to handle that.

**Comment by Marty850** [comment]
I was able to reproduce it on our test system with following steps:

1. stop DDF
2. start tuning services and wait a little bit
3. start DDF
4. Try to tune a statement

**Comment by mengxf123** [comment]
Hi @Marty850, thank you for your information. I will try on our environment and will back to you once I get any clue.

**Comment by mengxf123** [comment]
I can recreate the problem now, I will ask the team to see what happened and how to handle this problem. Thank you for reporting this issue. @Marty850

**Comment by mengxf123** [comment]
Hi Martin, the TMS server should not be appropriately started without the REPO(TMS meta DB) connection (for example, DDF stopped). But obviously, the current code does not handle it correctly.  And I tested it in our development release,and it didn't start the server as we wanted, with warnings:
```
WWKZ0022W: Application TuningService has not started in 30.000 seconds.                                                                               
CWWKE1102W: The quiesce operation did not complete. The server will now stop.                                                                          
CWWKE1105W: 1 runtime updates did not complete during the quiesce period.                                                                              
CWWKE1107W: 1 threads did not complete during the quiesce period.                                                                                      
CWWKE0036I: The server TMS-dev stopped after 1 minutes, 9.045 seconds.      
```
And in the log, there is also an error like:
`ERROR -- Failed to create connection(pool) to repo DB, ensure the repo configuration correct and restart server! `

**Comment by mengxf123** [comment]
Since the fix will be delivered in our next release, so we are not going to fix it in the current release unless there will be a PTF that needs to be released before the next release.  Does it make sense to you? 

Now, you may need to update the system automation process to ensure that TMS starts after DDF starts successfully. However, thank you very much for reporting on this issue.

**Comment by Marty850** [comment]
Hi @mengxf123, 
thats ok for us. We already updated the system automation process. 
Thank you very much!

**Comment by zhaoxq** [comment]
Fix is delivered in PTF, can you pls verify and close the issue?

**Comment by Marty850** [comment]
@zhaoxq Could you please tell me the PTF number. I ordered service and got UI81582 only. The description doesn't fit for that issue.

**Comment by mengxf123** [comment]
@Marty850 Hello Martin, I will confirm with team whether UI81852 has fixed this problem, and I will reply to you later. I thought this PTF was just fixing some security issues with our annual review. As I commented earlier, a fix for this issue will be released in our next release. Will double check and let you know later. Thank you!

**Comment by mengxf123** [comment]
@Marty850 Hi Martin, please keep this issue open and the fix will be included in our next release as I mentioned earlier. I will let you know when we ship the new release. Thanks.

**Comment by zhaoxq** [comment]
This is fixed in SQL Tuning Services 2.1.

**Comment by katsoohoo** [comment]
Closing this issue as the fix has been provided.
